docs/architecture/adr-040-storage-and-smt-state-commitments.md

```diff
@@ -206,6 +206,27 @@ Some objects may be saved with key, which contains a Protobuf message type. Such
 
 TODO: finalize this or move to another ADR.
 
+## Migration
+
+Using the new store will require a migration. 2 Migrations are proposed:
+1. Genesis export -- it will reset the blockchain history.
+2. In place migration: we can reuse `UpgradeKeeper.SetUpgradeHandler` to provide the migration logic:
+
+    ```go 
+app.UpgradeKeeper.SetUpgradeHandler("adr-40", func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+
+    storev2.Migrate(iavlstore, )
+
+    // RunMigrations returns the VersionMap
+    // with the updated module ConsensusVersions
+    return app.mm.RunMigrations(ctx, vm)
+})
+    ```
+
+    The `Migrate` function will read all entries from the save them to the AD-40 combined KV store. Cash layer should not be used and the operation must finish with a single Commit call.
+
+Inserting records to the `SC` (SMT) component should be the bottleneck. Unfortunately SMT doesn't support batch transactions. 
```

Comment:
Is this something we should consider adding support for?

Reply:
Let's firstly test it and then decide. If the performance will be OK for few major chains then we can push it to a later feature upgrade.

Reply:
Added a note that SMT batch transaction will be considered as a new future after the main release.